### ISSUE TITLE
querying aria-role superclass hierarchies

### ISSUE THREAD

**Issue by alexkrolick** [testing-library/dom-testing-library#448]
`getByRole` should be able to understand superclass hierarchies such as:

```
input      [abstract, debatable whether this should be queryable?]
  checkbox
    switch   (should be able to get switch by switch, checkbox, or input)
  option
  radio
  slider
  spinbutton
  textbox
```

Looks related to #446 

Examples: https://codesandbox.io/s/young-tree-68kbl

**Comment by kentcdodds** [comment]
Not sure whether to label this as a bug or enhancement, but either way, this would be great :)

**Comment by kentcdodds** [comment]
Actually, I feel like this should be something that aria-query or dom-accessibility-api should handle 🤔

It's definitely an enhancement though :)

**Comment by alexkrolick** [comment]
Sounds like aria-query provides this functionality

**Comment by delca85** [comment]
If it sounds good to all of you, I could try to make this enhancement!

**Comment by kentcdodds** [comment]
That sounds great to me @delca85!

**Comment by kentcdodds** [comment]
We've decided to not implement this.
